src/main/java/mekanism/common/tile/base/TileEntityMekanism.java

```diff
@@ -431,8 +438,46 @@ public void addComponent(ITileComponent component) {
     }
 
     @Nonnull
+    @Override
+    @SuppressWarnings("ConstantConditions")
     public Component getName() {
-        return TextComponentUtil.translate(Util.makeDescriptionId("container", getBlockType().getRegistryName()));
+        final var defaultName = TextComponentUtil.build(getBlockType());
+        if (isNameable()) {
+            return hasCustomName() ? getCustomName() : defaultName;
+        }
+        return defaultName;
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("ConstantConditions")
+    public Component getDisplayName() {
+        final var defaultName = TextComponentUtil.translate(Util.makeDescriptionId("container", getBlockType().getRegistryName()));
```

Comment:
Two things:
1. For now I think I am going to hold off in a big way from using var in Mekanism
2. The reason I was mentioning wrapping in isNameable, is not all tiles have this translation key now. Which is why I was saying to use a different fallback `TextComponentUtil.build(getBlockType())` for if it is not nameable, but use this one for if it is nameable and just doesn't have a custom name.